src/test/crimson/seastore/test_object_data_handler.cc

```diff
@@ -709,7 +709,52 @@ TEST_P(object_data_handler_test_t, random_overwrite) {
       logger().info("random_writes: {} done replaying/checking", i);
     }
     read(0, 4<<20);
-    unset_overwrite_threshold();
+    disable_delta_based_overwrite();
+  });
+}
+
+TEST_P(object_data_handler_test_t, overwrite_then_read_within_transaction) {
+  run_async([this] {
+    enable_delta_based_overwrite();
+    auto t = create_mutate_transaction();
+    write(*t, 0, 4096 * 3, 'a');
+    submit_transaction(std::move(t));
+    t = create_mutate_transaction();
+    write(*t, 4096, 4096, 'b');
+    read(*t, 1024, 4096 + 1024);
+    write(*t, 8192, 4096, 'c');
+    read(*t, 2048, 8192);
+    write(*t, 0, 4096, 'd');
+    write(*t, 4096, 4096, 'x');
+    submit_transaction(std::move(t));
+    read(1024, 8192 - 1024);
+    read(0, 4096 * 3);
+    restart();
+    epm->check_usage();
+
+    auto t1 = create_mutate_transaction();
+    write(*t1, 4096, 4096, 'e');
+    read(*t1, 4096, 4096);
+    auto t2 = create_read_transaction();
+    bufferlist bl = with_trans_intr(*t2, [&](auto &t) {
+      return ObjectDataHandler(MAX_OBJECT_SIZE).read(
+        ObjectDataHandler::context_t{
+          *tm,
+          t,
+          *onode
+        },
+        4096,
+        4096);
+    }).unsafe_get0();
+    bufferlist known;
+    known.append(
+      bufferptr(
+	known_contents,
+	4096,
+	4096));
+    EXPECT_EQ(bl.length(), known.length());
+    EXPECT_NE(bl, known);
```

Comment:
Let's extend the test to make sure the change also works with data extents at state `EXIST_MUTATION_PENDING`.

Reply:
@cyx1231st Can you please elaborate the scenario? I think that current implementation remove s EXIST_CLEAN extent and then insert a new extent before overwriting.

Reply:
I think the following steps might be possible to construct such scenario:
1. Create and commit an extent
2. In another transaction, read it out as clean, split/remap it into 2 smaller extents (EXIST_CLEAN)
3. Overwrite the remapped extent (EXIST_CLEAN)
4. Assert the extent is at state EXIST_MUTATION_PENDING
5. Commit the transaction
6. Do restart and validations

Let me know if the above steps won't work.

Reply:
Done. Can you take a look?